object-store-internal/src/lib.rs

```diff
@@ -507,73 +508,91 @@ impl PyObjectStore {
 
     /// Save the provided bytes to the specified location.
     #[pyo3(text_signature = "($self, location, bytes)")]
-    fn put(&self, location: PyPath, bytes: Vec<u8>) -> PyResult<()> {
-        self.rt
-            .block_on(self.inner.put(&location.into(), bytes.into()))
-            .map_err(ObjectStoreError::from)?;
-        Ok(())
+    fn put(&self, py: Python, location: PyPath, bytes: Vec<u8>) -> PyResult<()> {
+        py.allow_threads(|| {
+            self.rt
+                .block_on(self.inner.put(&location.into(), bytes.into()))
+                .map_err(ObjectStoreError::from)?;
+            Ok(())
+        })
```

Comment:
I always thought that if you didn't include the `py: Python` argument, that the GIL would be released automatically. Is this not true? To release the GIL you have to include the `py` argument **and** then also call `allow_threads`?

Reply:
I thought the same thing but that's not what my test showed. If you look at the docs: https://pyo3.rs/v0.21.1/parallelism#parallelism

It goes from

```rust
fn search_sequential(contents: &str, needle: &str) -> usize {
    contents.lines().map(|line| count_line(line, needle)).sum()
}
```

To

```rust
fn search_sequential_allow_threads(py: Python<'_>, contents: &str, needle: &str) -> usize {
    py.allow_threads(|| search_sequential(contents, needle))
}
```

So I think we are wrong.

I can confirm with @davidhewitt tomorrow.